src/Discord/Provider.php

```diff
@@ -61,6 +61,24 @@ protected function getUserByToken($token)
         return json_decode($response->getBody()->getContents(), true);
     }
 
+    /**
+     * @param array $user
+     *
+     * @return string|null
+     */
+    protected function formatAvatar(array $user)
+    {
+        if (is_null($user['avatar'])) {
+            return null;
+        }
+
+        $isGif = preg_match('/a_.+/m', $user['avatar']) === 1;
+        $extension = $this->getConfig('avatar_gifs', true) && $isGif ? 'gif' :
+            $this->getConfig('avatar_default_extension', 'webp');
+
+        return sprintf('https://cdn.discordapp.com/avatars/%s/%s.%s', $user['id'], $user['avatar'], $extension);
```

Comment:
Will this always work? Is it possible our dynamic URL could 404?

Reply:
As long as the default is either jpg, png or webp there shouldnt be any 404's.

Reply:
I can explain a little bit more in the readme if wanted

Reply:
No worries, `jpg` please

Could you add an `@see` docblock tag with a link to the discord docs?

Reply:
Would be good to document in the readme a bit more as well, thanks!

Reply:
> Could you add an `@see` docblock tag with a link to the discord docs?
Will add that now as well